### DISCUSSION TITLE
Web-ui password is always wrong in Nightly tag

### DISCUSSION THREAD

**Discussion by Filarius** [wg-easy/wg-easy#1181] [Q&A] [answered]
So I trying to set password using nightly version of wg-easy, but its always "wrong".

Password 12345

Making hash for Docker env

`python3 -c "import bcrypt; password = b'12345'; assert len(password) < 72, 'Password must be less than 72 bytes due to bcrypt limitation'; hashed = bcrypt.hashpw(password, bcrypt.gensalt()); print(f'The hashed password is: {hashed.decode()}'); docker_interpolation = hashed.decode().replace('$', '$'*2); print(f'The hashed password for a Docker env is: {docker_interpolation}')"`

```
The hashed password is: $2b$12$ZEYLPTfXb2D7u7.guUCf1e4QUyiYegqqoMGR8hlKTGSzObQYelz9K
The hashed password for a Docker env is: $$2b$$12$$ZEYLPTfXb2D7u7.guUCf1e4QUyiYegqqoMGR8hlKTGSzObQYelz9K
```
```
 docker run -d \
  --name=wg-easy \
  -e LANG=en \
  -e WG_HOST=qwer.ty \
  -e PASSWORD_HASH=$$2b$$12$$ZEYLPTfXb2D7u7.guUCf1e4QUyiYegqqoMGR8hlKTGSzObQYelz9K \
  -e PORT=51821 \
  -e WG_PORT=51820 \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
ghcr.io/wg-easy/wg-easy:nightly

```

**Comment by tetuaoro** [comment, marked as answer]
Try `PASSWORD_HASH='$2b$12$ZEYLPTfXb2D7u7.guUCf1e4QUyiYegqqoMGR8hlKTGSzObQYelz9K'` with single quote & remove double `$`

**Reply by Filarius** [reply]
Thanks, it works now.
Its only case i did not get to try. I tryed with double quote and single $
Also don't forget to clear webbrowser cache - i had problem like there no web page change after password. 
Seems webbrowser got old page from master tag as i got it try first before nighty version.

Is it about somebody need to update readme.md

**Reply by eandersons** [reply]
Another alternative, since you are running container from terminal with the `docker run` command, is to escape dollar signs by adding backslash before them (works with unquoted and double quoted values): `PASSWORD_HASH=\$2b\$12\$ZEYLPTfXb2D7u7.guUCf1e4QUyiYegqqoMGR8hlKTGSzObQYelz9K`.

A side note - escaping with double dollar sign works in Compose file, not in terminal.
